File: android/src/main/java/com/mattermost/networkclient/sessionattributes/SessionAttributes.kt
package com.mattermost.networkclient.sessionattributes

import android.content.Context

/**
 * React-free entry point for native code outside this library (e.g. app
 * background handlers, standalone OkHttp usage) to resolve the outbound
 * X-MM-Session-Attributes header for a server.
 *
 * Callers that reach this outside the React module (e.g. WorkManager jobs) must
 * call [init] from their Application.onCreate, which runs before any component
 * in the process.
 */
object SessionAttributes {
    @JvmStatic
    fun init(context: Context) {
        SessionAttributesEngine.init(context)
    }

    @JvmStatic
    fun getOutboundHeader(serverUrl: String): String? {
        return SessionAttributesEngine.getOutboundHeader(serverUrl)
    }
}

File: ios/SessionAttributes/SessionAttributes.swift
// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import Foundation

/// React-free entry point for native code outside this library (e.g. Gekidou,
/// app extensions, standalone URLSession/OkHttp usage) to resolve the outbound
/// `X-MM-Session-Attributes` header for a server.
@objc public class SessionAttributes: NSObject {
    @objc public static func getOutboundHeader(_ serverUrl: String) -> String? {
        return SessionAttributesEngine.shared.getOutboundHeader(serverUrl)
    }
}
